### PR TITLE
Remove required location property

### PR DESCRIPTION
**Pull request by CarsonF**
Update remaining places where location is passed in to ignore/handle it but use `useLocation` hook instead.

### CODE REVIEW

**Comment by fzaninotto** [comment]
Why should we use `useLocation` in components that are children of `Route` and therefore receive the `location` prop?

**Comment by CarsonF** [comment]
react-router is moving away from that API with the v5.1 release. It's stated that the `component` and `render` props on `Route` will go away with the v6 release. They expect you to use the hooks instead of them cloning the element to inject props.
https://reacttraining.com/blog/react-router-v5-1/

File: packages/ra-core/src/controller/useCreateController.ts [hunk 1]
```diff
@@ -39,8 +39,8 @@ export interface CreateProps {
     hasEdit?: boolean;
     hasList?: boolean;
     hasShow?: boolean;
-    location: Location;
-    match: Match;
+    location?: Location; // deprecated
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/useCreateController.ts]: it's not deprecated, it's ignored. react-router still injects these props.

File: packages/ra-core/src/controller/useListController.ts [hunk 2]
```diff
@@ -29,7 +30,7 @@ export interface ListProps {
     hasEdit?: boolean;
     hasList?: boolean;
     hasShow?: boolean;
-    location: Location;
+    location?: Location; // deprecated
```
Comment (fzaninotto) [inline packages/ra-core/src/controller/useListController.ts]: same

File: packages/ra-ui-materialui/src/list/List.js [hunk 3]
```diff
@@ -90,8 +90,8 @@ List.propTypes = {
     hasEdit: PropTypes.bool.isRequired,
     hasList: PropTypes.bool.isRequired,
     hasShow: PropTypes.bool.isRequired,
-    location: PropTypes.object.isRequired,
-    match: PropTypes.object.isRequired,
+    location: PropTypes.object, // deprecated
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/List.js]: same

File: packages/ra-ui-materialui/src/layout/NotFound.js [hunk 4]
```diff
@@ -49,13 +49,13 @@ const NotFound = ({
     className,
     classes: classesOverride,
     title,
-    location,
+    location, // not used, but don't want to pass to div
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/layout/NotFound.js]: prefer a `sanitizeRestProps` function that cleans up the props passed to the div.

File: packages/ra-ui-materialui/src/list/List.js (hunk 3 above)
Comment (CarsonF) [reply packages/ra-ui-materialui/src/list/List.js]: I'm not very good with PropTypes. Should I remove it? Won't that make React complain if it is passed in?

File: packages/ra-core/src/controller/useCreateController.ts (hunk 1 above)
Comment (CarsonF) [reply packages/ra-core/src/controller/useCreateController.ts]: Would you like me to remove it from the interface instead?

Comment (fzaninotto) [reply packages/ra-core/src/controller/useCreateController.ts]: no, just remove the `// deprecated` comment

File: packages/ra-ui-materialui/src/list/List.js (hunk 3 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/list/List.js]: Don't remove the line, just the comment.

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**